### PR TITLE
fix: TextArea warning with React 16.13

### LINKED ISSUE #21697: TextArea mix shorthand and non-shorthand properties for the same value.
### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/antd-reproduction-template-6n7oo)

### Steps to reproduce
Use `Input.TextArea` with `autoSize` prop.

### What is expected?
do not throw warning/error.

### What is actually happening?
throw warning/error: 
```
Removing a style property during rerender (overflow) when a conflicting property is set (overflowY) can lead to styling bugs. To avoid this, don't mix shorthand and non-shorthand properties for the same value; instead, replace the shorthand with separate values.
```

| Environment | Info |
|---|---|
| antd | 4.0.0 |
| React | react 16.13.0 |
| System | linux fedora 31 |
| Browser | ff v.73, chrome v.80 |

### PR DESCRIPTION
**Pull request by zombieJ**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

fix #21697

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Fix TextArea style get warning in `react@16.13`.       |
| 🇨🇳 Chinese |     修复 TextArea 样式在 `react@16.13` 下会报警告的问题。     |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/input/ResizableTextArea.tsx [hunk 1]
```diff
@@ -121,7 +121,11 @@ class ResizableTextArea extends React.Component<TextAreaProps, TextAreaState> {
     const style = {
       ...this.props.style,
       ...textareaStyles,
-      ...(resizeStatus === RESIZE_STATUS_RESIZING ? { overflow: 'hidden' } : null),
+      ...(resizeStatus === RESIZE_STATUS_RESIZING
+        ? // React will warning when mix `overflow` & `overflowY`.
+          // We need to define this separately.
+          { overflowX: 'hidden', overflowY: 'hidden' }
+        : null),
```
Comment (afc163) [inline components/input/ResizableTextArea.tsx]: 这能加用例么

File: components/input/__tests__/textarea.test.js [hunk 2]
```diff
@@ -46,6 +47,9 @@ describe('TextArea', () => {
     expect(mockFunc).toHaveBeenCalledTimes(2);
     wrapper.update();
     expect(wrapper.find('textarea').props().style.overflow).toBeFalsy();
+
+    expect(errorSpy).not.toHaveBeenCalled();
+    errorSpy.mockRestore();
```
Comment (afc163) [inline components/input/__tests__/textarea.test.js]: mockRestore 和 mockReset 有啥区别？

Comment (zombieJ) [reply components/input/__tests__/textarea.test.js]: `mockReset` 清空记录，但是继续监听。`mockRestore` 清空记录且停止监听。

**Comment by gotjoshua** [comment]
Will this end up on the 3.x tree in the next release?

**Comment by chenfengjw163** [comment]
这个问题3.x什么时候更新？

**Pull request merged**